### DISCUSSION TITLE
Increase available RAM for DivMMC

### DISCUSSION THREAD

**Discussion by davidpesqueirasouto** [UzixLS/zx-sizif-512#21] [Ideas]
Hi Eugene,

Is it possible to increase the available RAM for the DivMMC? In DivMMC mode only 384K of the 512K available on RAM chip are used, leaving 128K free. Is it possible to add them to the existing 128K to have a total of 256K in the DivMMC?

Thanks in advance

**Comment by UzixLS** [comment]
Hello! Rest 128K available via #DFFD port in Pentagon mode.

**Comment by tuFFo126** [comment]
How to use memory above 128k in pentagon mode?  I have demos for 256k and 512k but they don't start.

**Reply by UzixLS** [reply]
Which demos?

**Reply by tuFFo126** [reply]
256k
mirrormirrorpentagon256k.trd
Snow Buisness [Pentagon256k].trd
virus_demo.trd
512k
adpcm.scl
badapple.trd
butt512k.scl
CFM-512I.TRD
disgift.scl
LOVEDEMO.SCL

**Reply by tuFFo126** [reply]
How to use full 512kB RAM without SD card?  where do i download the programs from then?  Because when there is an SD card, 256kB RAM is available.  But those given Demos on 256k won't start anyway.

**Reply by UzixLS** [reply]
You can use BDI/DivIDE/DivMMC(untested) addon and load software using it

**Reply by UzixLS** [reply]
Also please note that Pentagon use #7ffd port for extra pages, but Sizif uses #dffd. #DFFD choice caused due to much better compatibility with old software.

**Reply by tuFFo126** [reply]
I don't have BDI.  I have DivIDE_57c from ByteDeligt but it freezes the computer.  I have divMMC (Just NANO SD) from Zaxon crashes my computer too.

**Reply by tuFFo126** [reply]
And can port 7ffdh be somehow simulated instead of dffdh?

**Reply by UzixLS** [reply]
There is no option for extra bits in 7ffd port currently.
SD card should be ejected from internal slot to use DivIDE/DivMMC addon

**Reply by tuFFo126** [reply]
the SD card is removed and yet for divide and divmmc the computer freezes

**Reply by UzixLS** [reply]
:(
Maybe something broke in the latest releases. I will сheck with my divide.

**Reply by tuFFo126** [reply]
![20230426_231835](https://user-images.githubusercontent.com/124886433/234706171-45985870-1a00-4d48-baf7-d64f9888a644.jpg)

**Reply by tuFFo126** [reply]
![20230426_232103](https://user-images.githubusercontent.com/124886433/234706047-15c51088-98af-4cc4-b1fa-c78adb2c09c8.jpg)

**Comment by tuFFo126** [comment]
trying to run anything freezes the computer.  magic doesn't work at all

**Comment by tuFFo126** [comment]
strange, it worked with divide, but when trying to use the magic button then running anything crashes the computer

**Reply by tuFFo126** [reply]
Ok divIDE works, but you can't use the magic button at all, it crashes the computer afterwards.
if you want to change the startup configuration, e.g. increase the clock frequency, you must do it before inserting the controller, set the "eprom" write jumper.  The only problem I noticed is when the sound booster is written to "eprom" it disappears from the sound booster configuration menu.  What are the three goldpins on the sound booster for, maybe you need to set some jumpers in it to save the configuration?

**Reply by UzixLS** [reply]
I checked my DivIDE and it's works fine, but you need to use NMI button on DivIDE, not on Sizif (I will see if it's possible to fix).
Are you using bob_fossil's LFN browser? It's require 128KB of RAM, but DivIDE have only 32KB. Default esxdos's NMI browser works fine.

**Reply by UzixLS** [reply]
https://cloud.err200.net/s/9FAyMHtMydTWxDs

**Reply by tuFFo126** [reply]
that's exactly how my divIDE works, I just used the magician button unnecessarily, I thought it shouldn't be used in programs.  So when we use the entire 512k memory, forget about changing the configuration, you have to do it beforehand and save it.

**Reply by tuFFo126** [reply]
I wanted to set a higher clock before inserting divIDE. I did it. However, inserting a divIDE restores the settings to the pre-write (default). After removing the divIDE, the saved settings return (of course, I put the controller in with the power off)
So it has to be?

**Reply by UzixLS** [reply]
That's how it should be. Setting are saved and loaded by Magic ROM. DivIDE shadows it, replacing by its own ROM.

**Reply by tuFFo126** [reply]
So using divIDE you can't use clocks higher than 3.5?  And in the case of BDI, how is it?

**Reply by UzixLS** [reply]
BDI does not shadow system ROM (only some addresses), so everything will work fine.

**Reply by tuFFo126** [reply]
Thank you for your comprehensive answers.

**Comment by tuFFo126** [comment]
Where can you get BDI?

**Reply by UzixLS** [reply]
I bought DIY kit on zx-pk.com forum

**Reply by tuFFo126** [reply]
![bd128mini](https://user-images.githubusercontent.com/124886433/234918791-e27a1657-cca3-46a2-8b43-102d1400e998.png)
Will this Beta Disk 128 mini work with ZX-Sizif 512k?

**Reply by UzixLS** [reply]
May be. I can't confirm as I don't have exactly this clone.

**Reply by tuFFo126** [reply]
As there is no SD card or divide from esxdos, from the pentagon menu, selecting the tr-dos option only causes a reset.  So the tr-dos rom is not built into sizif.  Your BDI must have tr-dos rom built in.  There are several versions of tr-dos in this Beta Disk 128 mini.  and there is also to use gluk menu you need to have BDI, hm probably BD128mini may not have?

**Reply by tuFFo126** [reply]
I bought it, we'll see how it works with sizif.  Works, among others  with zx-nucleon 512k.

**Reply by tuFFo126** [reply]
ATTENTION! Beta Disk 128 mini uses computer ports for its operation #31,63,95,127,255.
It cannot be combined with another device using the same port addresses, typically a Kempston joystick!
Could it be a problem to use with sizif.  How does your BDI work?

**Reply by UzixLS** [reply]
Thats ok if this addon asserts IORQGE signal on port address match.

**Reply by tuFFo126** [reply]
Full description:

BRAND NEW BETA DISK 128 MINI
BOXED VERSION OF THE MOST POPULAR FLOPPY CONTROLLER
FOR ZX NUCLEON & ZX SPECTRUM, FOUR TR-DOS SYSTEM VERSION AVAILABLE FOR USER

(New, miniaturized version)

(The device box is made by 3D printing. Please tolerate appearance imperfections, surface unevenness, etc., 3D printing cannot produce a perfect product)

(Complete, finished, functional device)


INTRODUCTION:

Beta Disk 128mini is a controller for 1 to 4 floppy disk drives for ZX Nucleon, ZX Spectrum and compatible computers. It comes from the original interface of the same name from British company named Technology Research Ltd, with which it is fully compatible. During the construction, maximum emphasis was placed on the smallest possible dimensions of the device, therefore the connection is simplified as much as possible by using CPLD chips and the interface does not have a through bus.

"128" in the interface name indicates that it is a version of the ZX Spectrum operating with both 48KB and 128KB RAM.

PARAMETERS:

• Optional to connect 1 to 4 floppy drives 3.5" or 5.25"
• Compatible with the ZX Nucleon 512KB, ZX Spectrum 48KB, 128KB „toast rack“ and 128K +2 Amstrad gray box. It is also compatible with other ZX Spectrum clones with a standard system connector and ROM compatible with ZX Spectrum 48KB or 128KB, for example Didaktik M. Didaktik Gama can be connected after additional fixing, see next description. Interface is incompatible with the ZX Spectrum +2A, +2B and +3, these have non-standard signals ROMCS (2x) on the edge connector and incompatible ROM.
• Use 5V only controller M5W1793 or analog MB8877A
• Four different versions of the system TR-DOS available (5.04T, 5.041, 5.05cz, 6.11Q), selection by button switches
• LED power-on indication (green LED) and TR-DOS system activation (red LED)
• The device is powered by the computer
• Standard 34-pin Shugart interface connector. ATTENTION, this is not an IBM bus standard!

COMPATIBILITY:

Because the schematic is largely taken from the original interface, is directly compatible with these
computers without any further modifications:
• ZX Nucleon 512KB
• ZX Spectrum 48KB
• ZX Spectrum 128KB „toast rack“ with radiator to the right, +2 Amstrad gray box
• Didaktik M

ZX Spectrum 128KB +2A, +2B and +3: This computers is incompatible in terms of the ROMCS signal (25B), which is doubled with them. In addition, they also contain incompatible BASIC ROM. Therefore, the Beta Disk 128C interface can not be used with them!

Didaktik Gama 80KB, 192KB: Didaktik Gama computers of all versions meets the compatibility of the system connector, but they have an internal 8255A parallel interface using the same port addresses (31, 63, 95, 127), as Beta Disk controller, you can not use this computer without any modifications with the Beta Disk! Otherwise, there will be a collision of the 8255A parallel interface with the controller 1793 on the bus, and they could be damaged!

ATTENTION! Beta Disk uses computer ports for its operation #31,63,95,127,255.
It cannot be combined with another device using the same port addresses, typically a Kempston joystick!

Link to documentation: Documentation for the HW will be sent by email upon request. It is a basic description, wiring diagram and drawing of the PCB assembly. It does not contain a detailed description of working with the controller through the TR-DOS system. The seller assumes that the customer has this knowledge, which can also be obtained from the Internet.

Best world software archive for Beta Disk: https://vtrd.in

**Reply by tuFFo126** [reply]
Unfortunately, "Beta Disk 128 mini CSS Electronics" does not work properly.  When in sizif there is kempston trying to read the disk there is a message "no disk".  The magic button works and you can configure sizif, when I switch the joystick to sinclair, you can select the drive *"b:" (I have such a cable and fdd sees drive B:" but when I call the CAT command, the message "error disk " and the sector with an error is shown.  The formatting attempt looks like it formats the disk but about a sector for about 30 seconds.It looks like there is a conflict. The controller is automatically detected as ZC/+3e (SD card)

**Reply by UzixLS** [reply]
That's sad. Unfortunately I have nor schematic/sources for this BDI variant, nor this BDI itself, so cant debug it :(
Are you working in Pentagon mode? Current CPLD firmware release have compatibility issues with addons in non-Pentagon mode (already fixed in master branch and will be released soon).

**Reply by tuFFo126** [reply]
I haven't noticed any issues with the addons I have.  I use these BD128mini in Pentagon mode because I wanted to use the full RAM.  My sizif was created a little over a month ago.  The zx-hd has problems with the synchronization of the displayed image.  Both dadanator and divide disable the magic button and sizif starts only in the startup configuration regardless of the configuration save.
Can you show a picture of your BDI from zx-pk.com?  I found materials there to make them myself, i.e. a colleague could try to make this controller based on them.

**Comment by tuFFo126** [comment]
I have files from the zx-pk.ru site on the basis of which you can make BDI-ZX, but there are several versions of pof files to upload to cpld (altera max).  What exactly do you have pcb version with which altera and controller?  could you show pictures of your copy?

**Reply by UzixLS** [reply]
<details>
  <summary>Photos</summary>

![IMG_20230511_142121](https://github.com/UzixLS/zx-sizif-512/assets/1556813/0a812ccb-d249-4498-a38b-d04f370dc3c0)

![IMG_20230511_142126](https://github.com/UzixLS/zx-sizif-512/assets/1556813/b71f753d-2497-4b37-b4bd-0d89ebe01c51)

</details>

**Reply by tuFFo126** [reply]
Thank you
